tests/e2e/stateapp/stateapp_test.go

```diff
@@ -415,3 +462,103 @@ func TestStateTransactionApps(t *testing.T) {
 		})
 	}
 }
+
+func TestMissingKeyDirect(t *testing.T) {
+	externalURL := tr.Platform.AcquireAppExternalURL(appName)
+	require.NotEmpty(t, externalURL, "external URL must not be empty!")
+
+	// This initial probe makes the test wait a little bit longer when needed,
+	// making this test less flaky due to delays in the deployment.
+	_, err := utils.HTTPGetNTimes(externalURL, numHealthChecks)
+	require.NoError(t, err)
+
+	for _, protocol := range []string{"http", "grpc"} {
+		t.Run(fmt.Sprintf("missing_key_%s", protocol), func(t *testing.T) {
+			stateURL := fmt.Sprintf("%s/test/%s/%s/statestore", externalURL, protocol, "get")
+
+			key := guuid.New().String()
+			request := newRequest(utils.SimpleKeyValue{key, nil})
+			body, _ := json.Marshal(request)
+
+			resp, status, err := utils.HTTPPostWithStatus(stateURL, body)
+
+			var states requestResponse
+			json.Unmarshal(resp, &states)
+
+			require.Equal(t, 200, status)
```

Comment:
shouldn't the response be no content(204) instead of ok(200)?

Reply:
204 is a no content and this technically did return some content. It's a JSON blob with a key that references no value. So it doesn't really mean much, but by status code standards I think is correct.

Reply:
I have this question asked in #2338. It's fine to ship this as is. If I end up changing the code to give a 204 here, I'll circle back and fix this test.